src/custom/utils/format.ts

```diff
@@ -17,10 +16,29 @@ export interface FormatSmartOptions {
   isLocaleAware?: boolean
 }
 
+function _buildSmallLimit(smallLimit: string | undefined, decimalsToShow: number): string {
+  if (smallLimit) {
```

Comment:
nit: maybe add some comment in the fn, so u explain what is it doing. 
I understand you want to dynamically generate the smallest amount for different currencies depending on their precision.  Good be good to add it, plus some examples (including that `1` case)